modules/core/src/main/java/org/apache/synapse/util/xpath/SynapseJsonPath.java

```diff
@@ -91,7 +98,22 @@ public SynapseJsonPath(String jsonPathExpression)  throws JaxenException {
         EIPUtils.setJsonPathConfiguration();
 
         this.contentAware = true;
-        this.expression = jsonPathExpression;
+        // supporting evaluation against a property
+        // Ex: json-eval($ctx:prop1.student.name)
+        Pattern fullPattern = Pattern.compile(JSON_PATH_WITH_PROP_REGEX);
```

Comment:
Shall we to a small test to compare this with Startswith()

Reply:
+1 fixed